python/langsmith/client.py

```diff
@@ -1754,12 +1854,24 @@ def update_run(
 
         if self._pyo3_client is not None:
             self._pyo3_client.update_run(data)
-        elif use_multipart and self.tracing_queue is not None:
-            # not collecting attachments currently, use empty dict
+        elif use_multipart:
             serialized_op = serialize_run_dict(operation="patch", payload=data)
-            self.tracing_queue.put(
-                TracingQueueItem(data["dotted_order"], serialized_op)
-            )
+            if self.compressed_runs_buffer is not None:
+                multipart_form = (
+                    serialized_run_operation_to_multipart_parts_and_context(
+                        serialized_op
+                    )
+                )
+                with self._buffer_lock:
+                    compress_multipart_parts_and_context(
+                        multipart_form, self.compressor_writer, self._boundary
+                    )
+                    self._run_count += 1
```

Comment:
I'd prefer to not track statefulness in this way if we can avoid it. Why isn't this tracked on the buffer?

Reply:
fixed